dockstore-webservice/src/main/java/io/dockstore/webservice/resources/proposedGA4GH/ToolsApiExtendedServiceImpl.java

```diff
@@ -217,6 +219,12 @@ public Response toolsIndexGet(SecurityContext securityContext) {
 
     @Override
     public Response toolsIndexSearch(String query, MultivaluedMap<String, String> queryParameters, SecurityContext securityContext) {
+        String unableToUseESMsg = "Could not use Elasticsearch search";
+        if (!ELASTIC_SEARCH_CONCURRENCY_LIMIT.tryAcquire(1)) {
+            LOG.error(unableToUseESMsg + ": too many concurrent requests.");
+            throw new CustomWebApplicationException("Could not use Elasticsearch search", HttpStatus.SC_INTERNAL_SERVER_ERROR);
+        }
+
```

Comment:
follow-up thought: don't lines 227 through 285 need to wrapped in a try ... finally? 
For example, if line 273 null pointers, does the semaphore ever get released?